parquet-column/src/main/java/parquet/vector/VectorizedColumnReaderImpl.java

```diff
@@ -554,32 +561,34 @@ public Void visit(DataPageV2 dataPageV2) {
     }
   };
 
+  /**
+   * read as many pages as necessary to fetch {@link parquet.vector.ColumnVector#DEFAULT_VECTOR_LENGTH} values
+   */
   private void readPages() {
-    int i = 0;
     long totalValuesRead = 0;
-    pages = new DataPage[MAX_VECTOR_LENGTH];
-    while(i < MAX_VECTOR_LENGTH) {
+    pages = new ArrayList();
+//    pages = new DataPage[MAX_VECTOR_LENGTH];
+    while(totalValuesRead < DEFAULT_VECTOR_LENGTH) {
       DataPage page = pageReader.readPage();
       if (page == null) {
         break;
       }
       page.accept(visitor);
-      pages[i] = page;
+      pages.add(page);
       readValues += page.getValueCount();
       totalValuesRead += page.getValueCount();
-      i++;
     }
 
-    if (i < MAX_VECTOR_LENGTH - 1) {
-      //couldn't read a complete batch, so get rid of nulls
-      //TODO make this more efficient
-      pages = Arrays.copyOf(pages, i);
-    }
+//    if (i < MAX_VECTOR_LENGTH - 1) {
```

Comment:
since I am now using an arraylist instead of an array. Previously I was using an array and I need to remove nulls so that the class processing it shouldn't deal with them. Now we are ok.